tools/static_word_checks.py

```diff
@@ -0,0 +1,67 @@
+import os
+import sys
+from typing import Any, Callable
+
+
+class StaticCheck:
+    """Run simple checks on the entire document or specific lines."""
+
+    def __init__(self):
+        self.directory = os.path.abspath(
+            os.path.join(__file__, "..", "..", "praw")
+        )
+        self.line_checks = []
+        self.full_file_checks = []
+
+    def add_line_check(
+        self, func: Callable[[str, int, str], Any]
+    ) -> Callable[[str, int, str], Any]:
+        self.line_checks.append(func)
+        return func
+
+    def add_full_file_check(
+        self, func: Callable[[str, str], Any]
+    ) -> Callable[[str, str], Any]:
+        self.full_file_checks.append(func)
+        return func
+
+    def run_checks(self):
+        status = 0
+        for current_directory, directories, filenames in os.walk(
+            self.directory
+        ):
+            for filename in filenames:
+                if not filename.endswith(".py"):
+                    continue
+                filename = os.path.join(current_directory, filename)
+                with open(filename, "r", encoding="utf-8") as fp:
```

Comment:
While it's nice to be explicit in many cases, I think we're consistent with just using `open(filename)` across this project for opening with the defaults. Could you please remove the latter arguments?

Reply:
What about the encoding? I know from experience that windows opens files in like cp-1252 so I keep the encoding because if there's any Unicode characters things might break.

Reply:
Can you give me an example? While I don't work within Windows often, I've never seen any such issue.

Reply:
https://stackoverflow.com/questions/42070668/python-3-default-encoding-cp1252

> There is a platform-dependent default encoding […] In many cases, but not all, the system default is UTF-8; you should never count on this default.
> 
> In other words, Python opens source files as UTF-8 by default, but any interaction with the filesystem will depend on the environment. It's strongly recommended to use open(filename, encoding='utf-8') to read a file.

Reply:
Thanks for the info. I think until we run into a problem. let's consistently not provide the encoding. Again, our files should really be ASCII anyway, so none of this should really matter.